firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -127,17 +127,27 @@ class FieldPath(object):
         parts: (one or more strings)
             Indicating path of the key to be used.
     """
-    pattern = re.compile(r'[A-Za-z_][A-Za-z_0-9]*')
+    simple_field_name = re.compile(r'[A-Za-z_][A-Za-z_0-9]*')
 
     def __init__(self, *parts):
         for part in parts:
-            if not isinstance(part, str):
-                raise ValueError("One or more components is not a string")
+            try:
```

Comment:
@tswast @lukesneeringer Is this kosher, or should the `six` module be involved here somewhere?

Reply:
Use of `str` should almost never occur (and likely **never** in `google-cloud-python`). Here it seems `six.string_types` should be used? But it's also possible that `six.text_type` is desired?

Reply:
I changed it to six.string_types.  Thought six might have had something like that, but a few days ago I didn't notice it on here: http://python-future.org/compatible_idioms.html and didn't check six and figured it wouldn't have been hard to write it that way.